autokeras/blocks/basic.py

```diff
@@ -853,32 +853,47 @@ class BertBlock(block_module.Block):
         clf = ak.AutoModel(inputs=input_node, outputs=output_node, max_trials=10)
     ```
     # Arguments
-        max_sequence_length: Int. The maximum length of a sequence that is
-            used to train the model.
+        max_sequence_length: Int or kerastuner.engine.hyperparameters.Choice.
+            The maximum length of a sequence that is used to train the model.
     """
 
     def __init__(
         self,
-        max_sequence_length: Optional[int] = None,
+        max_sequence_length: Optional[Union[int, hyperparameters.Choice]] = None,
         **kwargs,
     ):
         super().__init__(**kwargs)
-        self.max_sequence_length = max_sequence_length
+        self.max_sequence_length = utils.get_hyperparameter(
+            max_sequence_length,
+            hyperparameters.Choice(
+                "max_sequence_length", [128, 256, 512], default=128
+            ),
+            int,
+        )
 
     def get_config(self):
         config = super().get_config()
-        config.update({"max_sequence_length": self.max_sequence_length})
+        config.update(
+            {
+                "max_sequence_length": hyperparameters.serialize(
+                    self.max_sequence_length
+                )
+            }
+        )
         return config
 
+    @classmethod
+    def from_config(cls, config):
+        config["max_sequence_length"] = hyperparameters.deserialize(
+            config["max_sequence_length"]
+        )
+        return cls(**config)
+
     def build(self, hp, inputs=None):
         input_tensor = nest.flatten(inputs)[0]
 
-        max_sequence_length = self.max_sequence_length or hp.Choice(
-            "max_seq_len", [128, 256, 512], default=128
-        )
-
         tokenizer_layer = keras_layers.BertTokenizer(
-            max_sequence_length=max_sequence_length
+            max_sequence_length=self.max_sequence_length
```

Comment:
https://github.com/keras-team/autokeras/pull/1425/files#diff-7b757ddf51e45c0e02ad8de148d02fd2abd59d670f91d35cac71157e6684c13cR138
We need to use utils.add_to_hp here.

Reply:
Thanks @haifeng-jin